Source/WebCore/platform/graphics/Path.cpp

```diff
@@ -97,28 +97,28 @@ Path::Path(PathSegment&& segment)
     m_data = WTFMove(segment);
 }
 
-PathImpl& Path::setImpl(Ref<PathImpl>&& impl)
+Ref<PathImpl> Path::setImpl(Ref<PathImpl>&& impl)
 {
     auto& platformPathImpl = impl.get();
     m_data = WTFMove(impl);
     return platformPathImpl;
 }
 
-PlatformPathImpl& Path::ensurePlatformPathImpl()
+Ref<PlatformPathImpl> Path::ensurePlatformPathImpl()
```

Comment:
Usually we create a separate function with the "protected" prefix, so that a caller can decide whether to request a Ref or not.

Something like this:
```
Ref<PlatformPathImpl> Path::ensureProtectedPlatformPathImpl()
{
    return ensurePlatformPathImpl();
}
```